src/app/services/board/board-cache.service.spec.ts

```diff
@@ -80,4 +81,75 @@ describe('BoardCacheService', () => {
       expect(localStorage.get).toHaveBeenCalled();
     })();
   });
+
+  it('should be able to put a board into the cache and get it back out again', (done) => {
+
+    inject([BoardCacheService], (service: BoardCacheService) => {
+      spyOn(service, 'getCacheKey').and.returnValue('cache_test_only');
+
+      const boardSet = new OBZBoardSet();
+      boardSet.rootBoardKey = undefined;
+      const testBoardJSON = {
+        format: 'board_format',
+        id: 5,
+        locale: 'en_GB',
+        name: 'board_name',
+        description_html: '<b>desc</b>',
+        grid: {
+          rows: 2,
+          columns: 2,
+          order: [[1, null],
+          [null, 2]]
+        },
+        buttons: [
+          {
+            id: 1,
+            label: 'button1'
+          },
+          {
+            id: 2,
+            label: 'button2'
+          }
+        ],
+        images: [
+          {
+            id: 1,
+            url: 'http://example.com'
+          }
+        ],
+        sounds: [
+          {
+            id: 1,
+            url: 'http://another.com'
+          }
+        ]
+      };
+      const testBoard = new OBFBoard().deserialize(testBoardJSON);
+      boardSet.setBoard('test', testBoard);
+
+      service.retrieve().subscribe({
```

Comment:
Does this want to fail? or just warn?
Failure means the test can never get cleaned up because we don't then run the delete.
Warning means the test can never end up deleting real data that has a key clash (may be irrelevant due to domain scoping?)

Reply:
I think failing is legit